src/dataclients/InspectionClient.js

```diff
@@ -0,0 +1,61 @@
+import { handleErrors, API_URL } from 'utils/FetchUtils';
+
+
+/**
+ * API to fetch available Inspection Types
+ * 
+ */
+export const getInspectionParams = async () => {
+  return fetch(API_URL + 'fieldCropCycles/inspectionTypes',
+    {
+      headers: {
+        Authorization: 'Bearer ' + localStorage.getItem('authToken'),
+      }
+    })
+    .then(handleErrors)
+    .then((response) => response.json());
+};
+
+/**
+ * API to fetch Adhoc activities based on Inspection Types
+ * 
+ */
+export const getAdhocActivities = async (fieldCropCycleId, inspectionValues) => {
+  let qParam = "";
+  inspectionValues.forEach((iValue) => {
+    if (iValue.val === 0) {
+      qParam = qParam + iValue.code + "=false&"
+    }
+    if (iValue.val === 1) {
+      qParam = qParam + iValue.code + "=true&"
+    }
+  })
+  if (fieldCropCycleId === '' || qParam === "")
+    return [];
```

Comment:
The clients should not be responsible for doing these things. They should only make network calls and return the result. Please move this part of either the calling file or create a separate util file if required.